### PR TITLE
added model eval and tensorboard visualization for training 

### PR DESCRIPTION
**Pull request by ilankadar**

### CODE REVIEW

File: src/config/nexarear_squeezeDet_config.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def nexarear_squeezeDet_config():
 
   mc.IMAGE_WIDTH           = 1280
   mc.IMAGE_HEIGHT          = 720
-  mc.BATCH_SIZE            = 20
+  mc.BATCH_SIZE            = 1
```
Comment (roiaddi) [inline src/config/nexarear_squeezeDet_config.py]: I don't think we can't merge this

File: src/dataset/imdb.py [hunk 2]
```diff
@@ -44,6 +45,10 @@ def image_idx(self):
     return self._image_idx
 
   @property
+  def image_idx(self):
+    return self._test_image_idx
```
Comment (roiaddi) [inline src/dataset/imdb.py]: nice

File: src/dataset/nexarear.py [hunk 3]
```diff
@@ -41,6 +41,18 @@ def __init__(self, image_set, data_path, mc):
 
     self._eval_tool = './src/dataset/kitti-eval/cpp/evaluate_object'
 
+  def get_classes(self):
+    return self._classes
+
+  def get_test_set(self):
```
Comment (roiaddi) [inline src/dataset/nexarear.py]: don't you get it from the parent class?

File: src/dataset/nexarear.py [hunk 4]
```diff
@@ -72,8 +84,11 @@ def _load_nexarear_annotation(self):
     for img_fname in self._image_idx:
       boxes_fname = os.path.join(self._label_path, img_fname +'.json')
       if os.path.isfile(boxes_fname):
-        with open(boxes_fname) as infile:
-          boxes_in_image = json.load(infile)
+        try:
+          with open(boxes_fname) as infile:
+            boxes_in_image = json.load(infile)
+        except:
+          print ('Error while loading json file {}'.format(boxes_fname))
```
Comment (roiaddi) [inline src/dataset/nexarear.py:89]: logger.error/warn

File: src/train.py [hunk 5]
```diff
@@ -194,6 +205,28 @@ def train():
     tf.train.start_queue_runners(sess=sess)
 
     summary_writer = tf.summary.FileWriter(FLAGS.train_dir, sess.graph)
+    test_summary_writer = tf.summary.FileWriter(FLAGS.test_dir, sess.graph)
+
+    precision = tf.placeholder(tf.float32, name='precision')
```
Comment (roiaddi) [inline src/train.py]: can we move all of those to a function?

File: src/train.py [hunk 6]
```diff
@@ -247,6 +280,31 @@ def train():
       }
 
       if step % FLAGS.summary_step == 0:
+        test_set = imdb.get_test_set()
```
Comment (roiaddi) [inline src/train.py]: also here. let move this into a function. too much code in the main function

File: src/train.py [hunk 7]
```diff
@@ -280,19 +283,17 @@ def train():
...
         images_directory = imdb.get_images_path()
         if os.path.isdir(FLAGS.pred_json_folder):
             shutil.rmtree(FLAGS.pred_json_folder)
         os.makedirs(FLAGS.pred_json_folder)
-        print('Processing {} test images'.format(len(test_set)))
-        for test_image_basename in test_set:
-            test_image_fname = os.path.join(images_directory,test_image_basename)
-            compute_time, json_dict_res = infer_bounding_boxes(model, sess, test_image_fname, FLAGS.pred_json_folder)
-
+        print('Processing {} test images'.format(len(img_batch)))
```
Comment (roiaddi) [inline src/train.py]: logger

File: src/train.py [hunk 8]
```diff
@@ -142,12 +143,22 @@ def train():
...
+        assert FLAGS.net == 'squeezeDet' or FLAGS.net == 'squeezeDet+' or FLAGS.net == 'resnet50' or FLAGS.net == 'vgg16' or FLAGS.net == 'squeezeDet_nexarear', \
             'Currently only the squeezeDet model is supported for the NEXAREAR dataset'
         if FLAGS.net == 'squeezeDet':
           mc = nexarear_squeezeDet_config()
           mc.PRETRAINED_MODEL_PATH = FLAGS.pretrained_model_path
           model = SqueezeDet(mc, FLAGS.gpu)
+        elif FLAGS.net == 'squeezeDet_nexarear':
+          mc = nexarear_squeezeDet_config()
+          mc.LOAD_PRETRAINED_MODEL = False  # model parameters will be restored from checkpoint
+          checkpoint_file = '/opt/squeezeDet/data/model_checkpoints/squeezeDet_nexarear/model.ckpt-112000'
```
Comment (roiaddi) [inline src/train.py]: is this here by mistake?

File: src/train.py [hunk 9]
```diff
@@ -142,12 +143,22 @@ def train():
...
+          print(checkpoint_file)
+          model = SqueezeDet(mc, FLAGS.gpu)
+          saver = tf.train.Saver(model.model_params)
+          sess = tf.Session(config=tf.ConfigProto(allow_soft_placement=True))
+          saver.restore(sess, checkpoint_file)
+
         elif FLAGS.net == 'squeezeDet+':
```
Comment (roiaddi) [inline src/train.py]: is this the old config code?

File: src/train.py (hunk 8 above)
Comment (ilankadar) [reply src/train.py]: removed ( was added to test the the code)

**Comment by srippa-nexar** [comment]
Merging so that we can do refactoring. We will do through code review in the refactoring commit

File: src/trainx.py [hunk 10]
```diff
@@ -263,6 +421,14 @@ def train(mcfg):
     if os.path.isfile(args.cfg_file):
         mc = get_model_config(args.cfg_file)
 
+    if tf.gfile.Exists(mc.train.LOG_TRAIN_DIR):
```
Comment (roiaddi) [inline src/trainx.py:424]: check for existence and delete + makedirs should be in a function and not copy pasted

**Pull request merged**